### PR TITLE
[9.0][BKP][IMP] website_mail: optimise nonsense subscription

### PR DESCRIPTION
**Pull request by grindtildeath**
Backport of: https://github.com/odoo/odoo/commit/ef6f86dde4628b15aeb5879d268017c4a56f21fc

Same as:
 - 11.0: https://github.com/OCA/OCB/pull/1073
 - 10.0: https://github.com/OCA/OCB/pull/1078

### CODE REVIEW

File: addons/website_mail/controllers/main.py [hunk 1]
```diff
@@ -87,6 +87,13 @@ def website_message_subscribe(self, id=0, object=None, message_is_follower="on",
         _message_is_follower = message_is_follower == 'on'
         _object = request.registry[object]
 
+        record = _object.browse(cr, uid, _id, context=context).exists()
+        if not record:
+            return False
+
+        _object.check_access_rule(cr, uid, [_id], 'read', context)
+        _object.check_access_rule(cr, uid, [_id], 'read', context)
```
Comment (florentx) [inline addons/website_mail/controllers/main.py]: It looks like a typo in backport. Please fix.

**Review by florentx — CHANGES_REQUESTED** [review]

File: addons/website_mail/controllers/main.py (hunk 1 above)
Comment (grindtildeath) [reply addons/website_mail/controllers/main.py]: good catch :+1:

**Review by florentx — APPROVED** [review]

File: addons/website_mail/controllers/main.py [hunk 2]
```diff
@@ -87,6 +87,13 @@ def website_message_subscribe(self, id=0, object=None, message_is_follower="on",
         _message_is_follower = message_is_follower == 'on'
         _object = request.registry[object]
 
+        record = _object.browse(cr, uid, _id, context=context).exists()
+        if not record:
+            return False
+
+        _object.check_access_rights(cr, uid, 'read', context)
```
Comment (florentx) [inline addons/website_mail/controllers/main.py]: It does not fail, but it's wrong:

`    def check_access_rights(self, cr, uid, operation, raise_exception=True): # no context on purpose.`

**Review by florentx — CHANGES_REQUESTED** [review]

File: addons/website_mail/controllers/main.py (hunk 2 above)
Comment (grindtildeath) [reply addons/website_mail/controllers/main.py]: :facepalm: should be fixed now

**Review by florentx — APPROVED** [review]

**Review by gurneyalex — APPROVED** [review]

**Pull request merged**